src/main/java/io/vertx/core/http/Http2Settings.java

```diff
@@ -0,0 +1,340 @@
+  }
+
+  /**
+   * Set the {@literal SETTINGS_MAX_CONCURRENT_STREAMS} HTTP/2 setting
+   *
+   * @param maxConcurrentStreams the new value
+   * @return a reference to this, so the API can be used fluently
+   */
+  public Http2Settings setMaxConcurrentStreams(long maxConcurrentStreams) {
+    Arguments.require(maxConcurrentStreams >= Http2CodecUtil.MIN_CONCURRENT_STREAMS,
+        "maxConcurrentStreams must be >= " + Http2CodecUtil.MIN_CONCURRENT_STREAMS);
+    Arguments.require(maxConcurrentStreams <= Http2CodecUtil.MAX_CONCURRENT_STREAMS,
+        "maxConcurrentStreams must be < " + Http2CodecUtil.MAX_CONCURRENT_STREAMS);
+    this.maxConcurrentStreams = maxConcurrentStreams;
+    return this;
+  }
+
+  /**
+   * @return the {@literal SETTINGS_INITIAL_WINDOW_SIZE} HTTP/2 setting
+   */
+  public int getInitialWindowSize() {
+    return initialWindowSize;
+  }
+
+  /**
+   * Set the {@literal SETTINGS_INITIAL_WINDOW_SIZE} HTTP/2 setting
+   *
+   * @param initialWindowSize the new value
+   * @return a reference to this, so the API can be used fluently
+   */
+  public Http2Settings setInitialWindowSize(int initialWindowSize) {
+    Arguments.require(initialWindowSize >= Http2CodecUtil.MIN_INITIAL_WINDOW_SIZE,
+        "initialWindowSize must be >= " + Http2CodecUtil.MIN_INITIAL_WINDOW_SIZE);
+    this.initialWindowSize = initialWindowSize;
+    return this;
+  }
+
+  /**
+   * @return the {@literal SETTINGS_MAX_FRAME_SIZE} HTTP/2 setting
+   */
+  public int getMaxFrameSize() {
+    return maxFrameSize;
+  }
+
+  /**
+   * Set the {@literal SETTINGS_MAX_FRAME_SIZE} HTTP/2 setting
+   *
+   * @param maxFrameSize the new value
+   * @return a reference to this, so the API can be used fluently
+   */
+  public Http2Settings setMaxFrameSize(int maxFrameSize) {
+    Arguments.require(maxFrameSize >= Http2CodecUtil.MAX_FRAME_SIZE_LOWER_BOUND,
+        "maxFrameSize must be >= " + Http2CodecUtil.MAX_FRAME_SIZE_LOWER_BOUND);
+    Arguments.require(maxFrameSize <= Http2CodecUtil.MAX_FRAME_SIZE_UPPER_BOUND,
+        "maxFrameSize must be <= " + Http2CodecUtil.MAX_FRAME_SIZE_UPPER_BOUND);
+    this.maxFrameSize = maxFrameSize;
+    return this;
+  }
+
+  /**
+   * @return the {@literal SETTINGS_MAX_HEADER_LIST_SIZE} HTTP/2 setting
+   */
+  public int getMaxHeaderListSize() {
+    return maxHeaderListSize;
+  }
+
+  /**
+   * Set the {@literal SETTINGS_MAX_HEADER_LIST_SIZE} HTTP/2 setting
+   *
+   * @param maxHeaderListSize the new value
+   * @return a reference to this, so the API can be used fluently
+   */
+  public Http2Settings setMaxHeaderListSize(int maxHeaderListSize) {
+    Arguments.require(maxHeaderListSize >= Http2CodecUtil.MIN_HEADER_LIST_SIZE,
+        "maxHeaderListSize must be >= " + Http2CodecUtil.MIN_HEADER_LIST_SIZE);
+    this.maxHeaderListSize = maxHeaderListSize;
+    return this;
+  }
+
+  /**
+   * @return the extra settings used for extending HTTP/2
+   */
+  @GenIgnore
+  public Map<Integer, Long> getExtraSettings() {
```

Comment:
json object cannot use int as key.

that’s the main problem.

> On Apr 6, 2016, at 10:20 AM, Clement Escoffier <notifications@github.com> wrote:
> 
> In src/main/java/io/vertx/core/http/Http2Settings.java <https://github.com/eclipse/vert.x/pull/1353#discussion_r58667606>:
> 
> > +   *
> > +   * @param maxHeaderListSize the new value
> > +   * @return a reference to this, so the API can be used fluently
> > +   */
> > +  public Http2Settings setMaxHeaderListSize(int maxHeaderListSize) {
> > +    Arguments.require(maxHeaderListSize >= Http2CodecUtil.MIN_HEADER_LIST_SIZE,
> > +        "maxHeaderListSize must be >= " + Http2CodecUtil.MIN_HEADER_LIST_SIZE);
> > +    this.maxHeaderListSize = maxHeaderListSize;
> > +    return this;
> > +  }
> > +
> > +  /**
> > +   * @return the extra settings used for extending HTTP/2
> > +   */
> > +  @GenIgnore
> > +  public Map<Integer, Long> getExtraSettings() {
> Should it be converted to JsonObject to be polyglot ?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/eclipse/vert.x/pull/1353/files/c7a6a1ce8a1da75fc0edcccf686f25b82162b149#r58667606>